### ISSUE TITLE
git clean can't handle files that are pending-delete

### ISSUE THREAD

**Issue by rfmerrill** [git-for-windows/git#772]
In windows, if a file is deleted while it's open for reading by another process, and it's a full moon, and mercury is in retrograde, the file will become a "pending-delete" ghost file

$ touch logfile.txt
$ # here we run tail -f logfile.txt in another shell and keep it running
$ ls -l logfile.txt
-rw-rw-r--+ 1 rmerrill None 0 Jun  2 10:38 logfile.txt
$ git clean -fdx
Removing logfile.txt
$ ls -l logfile.txt
-rw-r----- 1 Unknown+User Unknown+Group 0 Jun  2 10:38 logfile.txt
$ git clean -fdx
fatal: Cannot lstat 'logfile.txt': Permission denied
$ # we kill the tail -f
$ ls -l logfile.txt
ls: cannot access 'logfile.txt': No such file or directory

Basically logfile.txt still shows up in ls (and one would assume readdir et al) while it's in pending delete state, but anything you try to do to it including lstat will fail, even if you are administrator.

This is especially annoying because it still happens even with --dry-run

This is git version 2.8.something I will update this with the full git --version output later, sorry.

Running on windows 7 64-bit in parallels.

**Comment by dscho** [comment]
I do not really think that we can do anything about this. Feel free to correct me if I am wrong. In the meantime, I'll close this ticket.
